packages/babel-helper-class-iife/src/index.js

```diff
@@ -0,0 +1,25 @@
+import { types as t } from "@babel/core";
+import annotateAsPure from "@babel/helper-annotate-as-pure";
+
+const classIIFEs = new WeakMap();
```

Comment:
I generally haven't been super comfortable with things that rely on being singletons like this because it relies on them being the same version and npm flattening things. Usually we've relied on flattening for better performance, but making flattening affect the actual transform output worries me a lot.

Reply:
good point, but without merging both class transforms (which I think @hzoo proposed numerous times) i have no idea how to achieve the desired goal of "pulling" stuff into a single IIFE here

Reply:
@loganfsmyth do you have an idea how I could tackle this one differently?

The only thing that comes to my mind is to wrap a class with IIFE in class-properties transform and hardcode some detection logic in transform-classes to check if it should create IIFE or if it should just inline output nodes in place (when it would know that it's already in the IIFE).

It would create indirect dependency on what class-properties does, but even if the detection logic fails it would just create an extra IIFE (so the output still would be valid, just not "optimized").